ql/src/java/org/apache/hadoop/hive/ql/stats/BasicStatsTask.java

```diff
@@ -136,6 +136,11 @@ public BasicStatsProcessor(Partish partish, BasicStatsWork work, HiveConf conf,
       }
     }
 
+    public boolean canTrim(){
```

Comment:
please javadoc on this method
I struggle to understand what does "trim" mean in this context, also, a huge amount of conditions inside need an explanation

Reply:
The stats update will work only for tables that are non-partitioned and in the queries where there is a single stats stage(otherwise it will lead to overcounting) and some other conditions. So here it checks for those conditions.